packages/components/Icon/src/Icon.types.ts

```diff
@@ -0,0 +1,35 @@
+import { SvgProps } from 'react-native-svg';
+export const iconName = 'Icon';
+
+export interface IFontIconProps
+{
+  fontFamily?: string;
+  fontSrcFile?: string;
+  codepoint: number;
+  fontSize?: number;
+  color?: string;
+}
+
+export interface ISvgIconProps
+{
+  uri?: string;
+  src?: React.FC<SvgProps>;
+  viewBox?: string;
+  color?: string;
+}
+
+export interface IRasterImageIconProps
+{
+  src: object;
+}
+
+export interface IIconProps
+{
+  ariaLabel?: string;
+  fontSource?: IFontIconProps;
+  svgSource?: ISvgIconProps;
+  rasterImageSource?: IRasterImageIconProps;
+  width?: number;
```

Comment:
I would expect width/height values to be set by some enum, similar to how persona coin does it. I would then expect those values to be transferred into the right values for svg/raster/font as appropriate.

Reply:
FluentUI Icon and ReactNative Image both allow direct specification of the width and height via numbers though. Shouldn't that be the expectation?

Reply:
Makes sense if we have prior art here.